bottle.py

```diff
@@ -550,14 +550,14 @@ def prepare(self):
 
     def all_plugins(self):
         """ Yield all Plugins affecting this route. """
-        unique = set()
-        for p in reversed(self.app.plugins + self.plugins):
-            if True in self.skiplist: break
-            name = getattr(p, 'name', False)
-            if name and (name in self.skiplist or name in unique): continue
-            if p in self.skiplist or type(p) in self.skiplist: continue
-            if name: unique.add(name)
-            yield p
+        if True in self.skiplist: return
+        skips = set(self.skiplist)
+        for plugins in self.plugins, self.app.plugins:
+            for p in reversed(plugins):
+                if p not in skips and type(p) not in skips:
+                    name = getattr(p, 'name', False)
```

Comment:
Do you really need `False` here ? `getattr` will return `None` by default.

Reply:
@oz123 `getattr()` raises `AttibuteError` by default!  So there needs to be a third argument, but IMHO it should be `None` and not `False`.

Reply:
@oz123, you wrong, if len(args) == 2 and obj has no attr, error rises:
```
getattr(None, 'name')
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/IPython/core/interactiveshell.py", line 3331, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-41470b2f07a7>", line 1, in <module>
    getattr(None, 'name')
AttributeError: 'NoneType' object has no attribute 'name'
```

Reply:
Yeah, my bad one. Mixed it `dict.get`!